services/database/mail.ts

```diff
@@ -0,0 +1,43 @@
+import prisma from "@database";
+
+/**
+ * Finds all associated parents and volunteers with a class with a restricted time range
+ * @param hoursWithin time range requirement, i.e., query the database for classes starting in hoursWithin hours
+ * @returns Promise<void> of a query described above
+ */
+export default async function findEmails(hoursWithin: number) {
+    const oneHourAfter = new Date();
+    const oneHourBefore = new Date();
+    oneHourAfter.setHours(oneHourAfter.getHours() + (hoursWithin + 1));
+    oneHourBefore.setHours(oneHourBefore.getHours() + (hoursWithin - 1));
+    const result = await prisma.class.findMany({
+        include: {
+            parentRegs: {
+                include: {
+                    parent: {
+                        include: {
+                            user: true,
+                        },
+                    },
+                },
+            },
+            volunteerRegs: {
+                include: {
+                    volunteer: {
+                        include: {
+                            user: true,
+                        },
+                    },
+                },
+            },
+        },
+        where: {
+            startDate: {
+                // restricting the startDate to be within hoursWithin +/- 1 hour from now
+                lte: oneHourAfter,
```

Comment:
I'm kinda confused why it's hoursWIthin +/- 1 haha
Also should one of these be not inclusive? Intuitively, i think if both ends of the range are inclusive you might get duplicates when something equals a range boundary

Also i kinda agree with Eric's comment on how to make sure you don't get duplicates?
Like if start time is 3 and hoursWithin is 1
At time 5, oneHourBefore is 5 + (hoursWithin - 1) = 5, oneHourAfter is 5 + (hoursWithin + 1) = 7, so if there were startDates with 7 then it'll be included
(since it runs every hour) at time 7, oneHourBefore is 7 and oneHourAfter is 8, then the startDate of 8 will get included again?

Reply:
Clarification:
Right now, hoursWithin +/- 1 means the range is 2h
I think if it runs every 1h, the range should be 1h (so theres no overlap)

I messaged Eric and we thought the bounds should be, >= hoursWithin and < hoursWithin + 1

like if rn the time is 0
and you have A=3, B=3.5, C=4
and hoursWithin is 3
Then for t=0, the range is >= 3 and < 4, you send A and B
for t=1, the range is >= 4 and < 5, you send C

Reply:
@victoranyun Should it be `lte` or `le`?